### ISSUE TITLE
local docker builder fails

### ISSUE THREAD

**Issue by coryschwartz** [ipfs/testground#476]
```
./testground run single  --builder=docker:go --runner=local:docker --instances 1 placebo/ok
Feb  5 18:14:22.328156	INFO	testground client initialized	{"addr": "localhost:8042"}
Feb  5 18:14:22.328386	INFO	testground client initialized	{"addr": "localhost:8042"}
fatal error from daemon: engine build error: unable to add replace directives to go.mod; exit status 2
```

This message comes from `pkg/build/docker.go`

I find that if I comment out everything related to the replaces so that `go mod edit` never runs, it works again. I commented out this whole section:

```go
	/*
		// If we have version overrides, apply them.
		var replaces []string
		for mod, ver := range in.Dependencies {
			// TODO(RK): allow to override target of replaces, so we can test against forks.
			replaces = append(replaces, fmt.Sprintf("-replace=%s=%s@%s", mod, mod, ver))
		}

			// Inject replace directives for the SDK modules.
			replaces = append(replaces,
				fmt.Sprintf("-replace=github.com/ipfs/testground/sdk/sync=../sdk/sync"),
				fmt.Sprintf("-replace=github.com/ipfs/testground/sdk/iptb=../sdk/iptb"),
				fmt.Sprintf("-replace=github.com/ipfs/testground/sdk/runtime=../sdk/runtime"))

			// Write replace directives.
			cmd := exec.CommandContext(ctx, "go", append([]string{"mod", "edit"}, replaces...)...)
			cmd.Dir = plandst
			_, err = cmd.CombinedOutput()
			if err != nil {
				return nil, fmt.Errorf("unable to add replace directives to go.mod; %w", err)
			}
	*/
```

if the go.mod file in every directory already includes replace directives, is this required?

**Comment by Stebalien** [comment]
Do you have go installed?

**Comment by Stebalien** [comment]
Could you try https://github.com/ipfs/testground/pull/483?

**Comment by coryschwartz** [comment]
You know what, the go installation is the culprit.

On opensuse tumbleweed, go version 1.12.15 is shipped by the package manager. I have go 1.13.6 installed, but it isn't the default in my PATH.

When I encountered this error, testground was built with go 1.13, but the daemon was running with go 1.12 in the PATH.

I got rid of the  just by putting go 1.13 farther up in the path and pruning pruning the old docker image.
